ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/compatibility/Utils.kt

```diff
@@ -23,10 +23,10 @@ internal suspend fun commonFetch(
         controller.abort()
     }
 
-    val promise: Promise<org.w3c.fetch.Response> = if (PlatformUtils.IS_BROWSER) {
-        fetch(input, init)
-    } else {
+    val promise: Promise<org.w3c.fetch.Response> = if (PlatformUtils.IS_NODE) {
```

Comment:
As described in my bug report, it would be even better if you had a

```kotlin
when (PlatformUtils.platform) {
    Node -> ...
    Browser -> ...
}
```

Then the logic for this decision would live in exactly one place and the next developer won't be able to re-introduce this bug by checking `IS_BROWSER` first again.